core/src/main/scala/org/apache/spark/status/api/v1/ExecutorListResource.scala

```diff
@@ -28,9 +28,11 @@ private[v1] class ExecutorListResource(ui: SparkUI) {
   @GET
   def executorList(): Seq[ExecutorSummary] = {
     val listener = ui.executorsListener
-    val storageStatusList = listener.storageStatusList
-    (0 until storageStatusList.size).map { statusId =>
-      ExecutorsPage.getExecInfo(listener, statusId)
+    listener.synchronized {
```

Comment:
I don't think it is necessary, essentially you are imposing that even the GET operations are to be processed sequentially....